src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, ViewChild, OnInit, Output, EventEmitter, Input, OnChanges } from '@angular/core';
+
+import { AuthenticationService } from '../../auth/authentication.service';
+import { Broadcaster } from '../../shared/broadcaster.service';
+import { Logger } from '../../shared/logger.service';
+
+import { WorkItem }          from '../../models/work-item';
+import { WorkItemService }   from '../work-item.service';
+import { IterationModel }    from '../../models/iteration.model';
+import { IterationService }  from '../../iteration/iteration.service';
+
+
+@Component({
+  selector: 'fab-planner-associate-iteration-modal',
+  templateUrl: './work-item-iteration-association-modal.component.html',
+  styleUrls: ['./work-item-iteration-association-modal.component.scss']
+})
+export class FabPlannerAssociateIterationModalComponent implements OnInit, OnChanges {
+
+  @Input() workItem: WorkItem;
+  @ViewChild('dropdownButton') dropdownButton: any;
+  @ViewChild('iterationAssociationModal') iterationAssociationModal: any;
+
+  iterations: IterationModel[];
+  selectedIteration: IterationModel;
+
+  constructor(
+    private auth: AuthenticationService,
+    private broadcaster: Broadcaster,
+    private workItemService: WorkItemService,
+    private logger: Logger,
+    private iterationService: IterationService,
+  ) {}
+
+  ngOnInit() {
+    this.iterations = this.iterationService.iterations;
+  }
+
+  ngOnChanges() {
+
+  }
+
+  onChangeIteration(iteration: IterationModel): void {
+    // Cache changed iteration
+    this.selectedIteration = iteration;
+
+    // Update dropdown button content
+    this.dropdownButton.nativeElement.innerHTML = this.selectedIteration.attributes.name + ' <span class="caret"></span>';
+
+    // Enable/disable confirmation button if iteration was changed
+    let commit = document.querySelector("#associate-iteration-button");
+    if (this.workItem.relationalData.iteration.id === this.selectedIteration.id) {
+      commit.setAttribute("disabled", "true");
```

Comment:
This is about detecting if the user has *changed* iteration. Only then enable the button and allow putting the change request to the API.

This particular condition catches:
1. WI has "iteration 1" as the associated iteration
2. User chooses "iteration 2" (button is enabled to change iteration)
3. But then the user switches back to "iteration 1"
4. **Now, because the associated and selected iterations are same - the button is disabled.**
5. Hence, it saves associating WI with an iteration it's already associated with (extra http-req).

Now, I understand your point (that don't show the current in the list iteration at all) but
1. Now you have an array-gymnastics to work with on each view update (which can regress horribly in future depending on slightest of model change)
2. As a user, I get paranoid when I can't set the old settings back in a control panel. Having the option to reset when I didn't meant to commit to a choice (and be sure no changes were applied, as the button goes disabled) is quite relieving.